### PR TITLE
04-playing-with-hangman

### PR DESCRIPTION
**Pull request by amyers2**
Anthony Myers
46989531

Note:  I am a 7345 student, but I attempted the 8k version of the assignment.

### CODE REVIEW

File: dictionary/README.md [hunk 1]
```diff
@@ -3,3 +3,6 @@ Please write a couple of sentences explaining your choice of restart
 strategy.
 
 
+I chose a restart strategy of :permanent, for the fact that each new
```
Comment (pragdave) [inline dictionary/README.md:6]: # +2

I don't normally give bonuses for the correct answer, but you're one of only three people who did, and it just makes me so happy...

File: dictionary/lib/dictionary.ex [hunk 2]
```diff
@@ -1,5 +1,11 @@
 defmodule Dictionary do
-
-  defdelegate random_word(), to: Dictionary.WordList
-
+  use Application
```
Comment (pragdave) [inline dictionary/lib/dictionary.ex:2]: No deduction, but I personally prefer to have the application start up in a separate module (Dictionary.Application) that contains an inline supervisor start-up.

File: dictionary/lib/dictionary/agent.ex [hunk 3]
```diff
@@ -0,0 +1,15 @@
+defmodule Dictionary.Agent do
```
Comment (pragdave) [inline dictionary/lib/dictionary/agent.ex:1]: A thing of beauty...

File: hangman/lib/hangman/game_listener.ex [hunk 4]
```diff
@@ -0,0 +1,31 @@
+defmodule Hangman.GameListener do
```
Comment (pragdave) [inline hangman/lib/hangman/game_listener.ex:1]: A couple of people did something like this.

Why not just use the fact that new_game returns a pid, and the client can pass that pid into Hangman.make_move and .tally, and the beam will work out where to send it to?

**Comment by pragdave** [comment]
This is really nice code—made me very happy :)

I'd be interested to hear if I misunderstood the listener stuff.

File: hangman/lib/hangman/game_listener.ex (hunk 4 above)
Comment (amyers2) [reply hangman/lib/hangman/game_listener.ex:1]: I guess I'm not fully understanding what you are suggesting.  Could you give a code example?

Comment (pragdave) [reply hangman/lib/hangman/game_listener.ex:1]: Each HumanPlayer client could look up the node running the game, then find the supervisor process by name. It then does something like this:

~~~ elixir
  defp new_game() do
    Node.connect(@hangman_server)
    :rpc.call(@hangman_server,
      Hangman,
      :new_game,
      [])
  end
~~~

This returns the pid of the game server on that node, and you ca then just call `Hangman.make_move(pid, "x")`

**Pull request closed without merging**